### PR TITLE
v 0.0.1

### PR DESCRIPTION
**Pull request by abramsymons**
- improve switching
- use stakes to evaluate each node share

### CODE REVIEW

File: node/routes.py [hunk 1]
```diff
@@ -97,10 +95,10 @@ def post_dispute() -> Response:
             "signature": utils.eth_sign(f'{zconfig.SEQUENCER["id"]}{timestamp}'),
         }
         return success_response(data=data)
-
-    zdb.init_txs(req_data["app_name"], req_data["txs"])
+    if req_data['app_name'] != '':
+        zdb.init_txs(req_data["app_name"], req_data["txs"])
     return error_response(ErrorCodes.ISSUE_NOT_FOUND)
-
+
```
Comment (abramsymons) [inline node/routes.py:101]: is this extra whitespace?

File: node/tasks.py [hunk 2]
```diff
@@ -80,7 +80,10 @@ def send_app_txs(app_name: str) -> None:
         )
     except Exception:
         zlogger.exception("An unexpected error occurred:")
-        zdb.add_missed_txs(app_name=app_name, txs=initialized_txs)
+        if len(initialized_txs) != 0:
+            zdb.add_missed_txs(app_name=app_name, txs=initialized_txs)
+            return
```
Comment (abramsymons) [inline node/tasks.py:85]: Should this `return` not be done after setting `is_sequencer_down` to `true`?
or moving `zdb.is_sequencer_down = True` to above `if`?

File: node/tasks.py [hunk 3]
```diff
@@ -80,7 +80,10 @@ def send_app_txs(app_name: str) -> None:
...
+        zdb.is_sequencer_down = True
```
Comment (abramsymons) [inline node/tasks.py:86]: when does this flag set to `False`?
I think after each successful sending tx to sequencer it should be set to `False`

File: node/routes.py [hunk 4]
```diff
@@ -86,8 +85,7 @@ def post_dispute() -> Response:
 
     if req_data["sequencer_id"] != zconfig.SEQUENCER["id"]:
         return error_response(ErrorCodes.INVALID_SEQUENCER)
-
-    if zdb.has_missed_txs():
+    if zdb.has_missed_txs() or (req_data['app_name'] == '' and zdb.is_sequencer_down):
```
Comment (abramsymons) [inline node/routes.py:88]: what does empty app name mean?

File: node/routes.py [hunk 5]
```diff
@@ -97,10 +95,10 @@ def post_dispute() -> Response:
             "signature": utils.eth_sign(f'{zconfig.SEQUENCER["id"]}{timestamp}'),
         }
         return success_response(data=data)
-
-    zdb.init_txs(req_data["app_name"], req_data["txs"])
+    if req_data['app_name'] != '':
```
Comment (abramsymons) [inline node/routes.py:98]: if this `if` is related to heartbeat txs, it should be removed.

File: node/routes.py (hunk 4 above)
Comment (mchitgarha78) [reply node/routes.py:88]: Yes, an empty app name with an empty list of txs does not get stored in the database.

File: node/routes.py (hunk 5 above)
Comment (mchitgarha78) [reply node/routes.py:98]: It has been removed in the last issue in switching branch.

File: node/routes.py (hunk 1 above)
Comment (mchitgarha78) [reply node/routes.py:101]: Fixed in this [commit](https://github.com/zellular-xyz/zsequencer/commit/ee5f0e7ebd51bb68f3c8d9b84186f11d1a58df64)

File: node/tasks.py (hunk 2 above)
Comment (mchitgarha78) [reply node/tasks.py:85]: Fixed in this [commit](https://github.com/zellular-xyz/zsequencer/commit/ee5f0e7ebd51bb68f3c8d9b84186f11d1a58df64)

File: node/tasks.py (hunk 3 above)
Comment (mchitgarha78) [reply node/tasks.py:86]: It has been set to `False` before the node wants to send initialized transactions or heartbeat.

**Review by abramsymons — COMMENTED** [review]
Thank you

**Pull request merged**